### PR TITLE
fix(network): app & component labels for network objects

### PR DESCRIPTION
**Pull request by mwangggg**
# Welcome to Cryostat! 👋
## Before contributing, make sure you have:
* [x] Read the [contributing guidelines](https://github.com/cryostatio/cryostat/blob/main/CONTRIBUTING.md)
* [x] Linked a relevant issue which this PR resolves
* [x] Linked any other relevant issues, PR's, or documentation, if any
* [x] Resolved all conflicts, if any
* [x] Rebased your branch PR on top of the latest upstream `main` branch
* [x] Attached at least one of the following labels to the PR: `[chore, ci, docs, feat, fix, test]`
* [x] [Signed all commits](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits): `git commit -S -m "YOUR_COMMIT_MESSAGE"`
_______________________________________________

Fixes: #607

### CODE REVIEW

**Review by tthvo — COMMENTED** [review]
Looks good! Thanks for handling this. 

You will also need to update the expected resource in tests. 

https://github.com/cryostatio/cryostat-operator/blob/2d386930dc96f0dcaf937987ec35874006c53b61/internal/test/resources.go#L2080

**Comment by tthvo** [comment]
While we are here, I think we should apply these changes to managed Services too.

**Comment by tthvo** [comment]
And Ingress too.

File: internal/controllers/services.go [hunk 1]
```diff
@@ -246,7 +246,8 @@ func (r *Reconciler) createOrUpdateService(ctx context.Context, svc *corev1.Serv
 	config *operatorv1beta1.ServiceConfig, delegate controllerutil.MutateFn) error {
 	op, err := controllerutil.CreateOrUpdate(ctx, r.Client, svc, func() error {
 		// Update labels and annotations
-		svc.Labels = config.Labels
+		mergeLabelsAndAnnotations(&svc.ObjectMeta, config.Labels, config.Annotations)
+
 		svc.Annotations = config.Annotations
```
Comment (tthvo) [inline internal/controllers/services.go:251]: Remove this?

File: internal/controllers/ingresses.go [hunk 2]
```diff
@@ -100,8 +100,8 @@ func (r *Reconciler) createOrUpdateIngress(ctx context.Context, ingress *netv1.I
 	config *operatorv1beta1.NetworkConfiguration) (*netv1.Ingress, error) {
 	op, err := controllerutil.CreateOrUpdate(ctx, r.Client, ingress, func() error {
 		// Set labels and annotations from CR
-		ingress.Labels = config.Labels
-		ingress.Annotations = config.Annotations
+		mergeLabelsAndAnnotations(&ingress.ObjectMeta, config.Labels, config.Annotations)
```
Comment (tthvo) [inline internal/controllers/ingresses.go:103]: I think we need to be extra careful here as `config.Labels` and `config.Annotations` can be `nil`. We will need the same handlers as in Services and Routes.

https://github.com/cryostatio/cryostat-operator/blob/2d386930dc96f0dcaf937987ec35874006c53b61/internal/controllers/services.go#L228

File: internal/test/resources.go [hunk 3]
```diff
@@ -2073,7 +2076,10 @@ func (r *TestResources) NewGrafanaRoute() *routev1.Route {
 func (r *TestResources) NewCustomGrafanaRoute() *routev1.Route {
 	route := r.NewGrafanaRoute()
 	route.Annotations = map[string]string{"grafana": "annotation"}
-	route.Labels = map[string]string{"grafana": "label"}
+	route.Labels = map[string]string{
+		"grafana":   "label",
+		"app":       r.Name,
+		"component": "cryostat"}
```
Comment (tthvo) [inline internal/test/resources.go:2082]: Lets put all ending bracket of the map to a newline? Just looks nicer!

**Review by tthvo — COMMENTED** [review]
Code looks good. Checking on clusters now..

File: internal/controllers/ingresses.go [hunk 4]
```diff
@@ -117,6 +117,43 @@ func (r *Reconciler) createOrUpdateIngress(ctx context.Context, ingress *netv1.I
 	return ingress, nil
 }
 
+func configureCoreIngress(cr *model.CryostatInstance) *operatorv1beta1.NetworkConfiguration {
```
Comment (tthvo) [inline internal/controllers/ingresses.go:120]: Ahh I forgot. These functions are just declared here. They need to be called:

https://github.com/cryostatio/cryostat-operator/blob/3fe4b93719b322972edfb45982745a5a3efef4fd/internal/controllers/ingresses.go#L46

https://github.com/cryostatio/cryostat-operator/blob/3fe4b93719b322972edfb45982745a5a3efef4fd/internal/controllers/ingresses.go#L69

Comment (tthvo) [reply internal/controllers/ingresses.go:120]: That would be mean another set of tests in Ingress to work on.

**Review by ebaron — COMMENTED** [review]
Hi @mwangggg, great work! I tried it out on OpenShift and Kind and everything seems to work as expected.

I thought of one small thing we could do to improve the tests. What if we add some different `app` and `component` labels to these Cryostat custom resources below? Then the tests will make sure that the operator overrides any `app` or `component` labels requested by the user with those required by the operator.

https://github.com/cryostatio/cryostat-operator/blob/5c3769f5a2823847214528fb655b297d688dac14/internal/test/resources.go#L322-L342

**Review by ebaron — APPROVED** [review]
Looks good! Thanks @mwangggg!

**Pull request merged**